extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -68,6 +69,53 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(StateProperty.PAGE_SIZE, (pageSizeState) => {
+      this.shouldTagFlip_(pageSizeState);
+    });
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @param {!Object} pageSize
+   * @private
+   */
+  shouldTagFlip_(pageSize) {
+    /* We only check the right hand side, as resizing only expands the border to the right. */
+
+    const storyPageWidth = pageSize.width;
+
+    this.measureElement(() => {
+      /*
+       * We are using offsetLeft and offsetWidth instead of getLayoutBox() because
+       * the correct measurements are not taken into account when using a CSS transform (such as translate),
+       * which we are using in the i-amphtml-amp-story-shopping-tag-inner-flipped class.
+       */
+      const {offsetLeft, offsetWidth} = this.element;
+
+      const shouldFlip = offsetLeft + offsetWidth > storyPageWidth;
+
+      const dotEl = this.shoppingTagEl_.querySelector(
```

Comment:
That is how it was originally. I was keeping it there to avoid doing unnecessary calculations in the mutate element - I'll make your change, but if anyone has any objections feel free to re-open and comment.